posix/clock_nanosleep.c

```diff
@@ -0,0 +1,80 @@
+/* High-resolution sleep with the specified clock.
+   Copyright (C) 2000-2012 Free Software Foundation, Inc.
+   This file is part of the GNU C Library.
+
+   The GNU C Library is free software; you can redistribute it and/or
+   modify it under the terms of the GNU Lesser General Public
+   License as published by the Free Software Foundation; either
+   version 2.1 of the License, or (at your option) any later version.
+
+   The GNU C Library is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+   Lesser General Public License for more details.
+
+   You should have received a copy of the GNU Lesser General Public
+   License along with the GNU C Library; if not, see
+   <http://www.gnu.org/licenses/>.  */
+
+#include <assert.h>
+#include <errno.h>
+#include <time.h>
+#include <sys/time.h>
+
+__typeof__(clock_nanosleep) __clock_nanosleep;
+
+/* This implementation assumes that these is only a `nanosleep' system
+   call.  So we have to remap all other activities.  */
+int
+__clock_nanosleep (clockid_t clock_id, int flags, const struct timespec *req, struct timespec *rem)
+{
+    struct timespec now;
+
+    if (__builtin_expect (req->tv_nsec, 0) < 0 || __builtin_expect (req->tv_nsec, 0) >= 1000000000){
+        return EINVAL;
+    }
+
+    /* If we got an absolute time, remap it.  */
+    if (flags == TIMER_ABSTIME)
```

Comment:
Is this intended to be `==` and not `&` ?

Reply:
I think flags must be equal to TIMER_ABSTIME but I could be wrong.

Following [this implementation](https://code.videolan.org/ePirat/vlc-legacy/-/merge_requests/16/diffs?commit_id=d50284da08a09481e50918549bdf4b8becb2caa0#7baa2427e25a0587c2e41e1663991e266131b7b5_0_70) for darwin and [this implementation](https://github.com/apc-llc/glibc-2.17/blob/master/sysdeps/unix/clock_nanosleep.c#L65) from glibc it should be "=="

Newer libc seem to use some nanosleep syscall.